### DISCUSSION TITLE
airflow fargate

### DISCUSSION THREAD

**Discussion by Lee2532** [apache/airflow#35923] [Q&A] [answered]
I'm going to run Fargate using airflow Kubernetes pod operator.

However, after the pod runs on the worker in the same node, the pod runs Fargate. I want to run it right away with Fargate.
what should I do?

**Comment by potiuk** [comment]
Wait until someone implements Fargate executor. I believe AWS team has it in their roadmap.

**Reply by Lee2532** [reply]
the first pod works over my node and that node pod running on fargate
result my airflow working two pod. is it right?

**Reply by potiuk** [reply, marked as answer]
If you start and wait for fargate using K8s POD operator that is not deferrable - yes. 

You have of course other options 

K8S POD operator works by runnign a separate POD for each task you are running. There are multiple options how it can be done differently - you can run Fargate operator (if there is one) -in either K8S or Celery Executor. Or you can employ Triggerer and Deferrable operators (if you have Fargate Deferrable operator ready - you'd need to look for the) or you could even have a pair of "Launch fargate" Operator and "Wait for Fargate" poke-reschedule sensor that will wake up periodically and only take process in celery or Pod in Kubernetes for the time of checking if fargate process completed.

So it's all in your hands how you deploy and build your DAGs, generally there are quite a few options you can follow to avoid it.

I don't believe there is Fargate Executor written yet by anyone - such that it would run natively tasks of Airflow. You need to use one of the existing executors to run Fargate from it (and then you have all the option above to decrease resource usage for that if you want).

But as far as I know MWAA team works on native AWS executor - and Fargate Executor was one of the options considered.
